### PR TITLE
export formatters functions

### PR DESCRIPTION
**Pull request by m7pr**
Part of https://github.com/insightsengineering/coredev-tasks/issues/447

### CODE REVIEW

File: R/formatters_var_labels.R
Comment (chlebowa) [inline R/formatters_var_labels.R:1]: Please add a @source or @author to indicate that these functions have been lifted from `formatters`.

File: NAMESPACE [hunk 1]
```diff
@@ -3,6 +3,9 @@
 export(basic_table_args)
 export(clean_brushedPoints)
 export(draggable_buckets)
+export(formatters_var_labels)
+export(formatters_var_relabel)
+export(formatters_with_label)
```
Comment (gogonzo) [inline NAMESPACE:8]: `var_labels<-` function is missing.

**Comment by gogonzo** [comment]
Is it okey to have similar functions in different packages? In `teal.data` we have `data_label` (to add label attribute to data) and `var_labels` here.
What about shortening name from `formatters_var_labels` to `varlabels`?

File: NAMESPACE (hunk 1 above)
Comment (m7pr) [reply NAMESPACE:9]: yeah, I was wondering if we need that. thanks for pointing this up for teal.transform https://github.com/insightsengineering/teal.transform/pull/148#discussion_r1267642939
Just exported for teal.widgets in this commit 6d9fa1737e9f6d5a4eb07002a488d5a5469b542f

File: R/formatters_var_labels.R
Comment (m7pr) [reply R/formatters_var_labels.R:1]: good idea c2541af5a257f6d5ff4ecee9301974ae3c738126 - I tried to prepend with `formatters_` to make it visible, but @source is more clear

File: R/formatters_var_labels.R [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+#' @seealso [formatters_var_labels()] [formatters_var_relabel()] [`formatters_var_labels<-`]
+#'
+#' @export
+#' @return \code{x} labeled by \code{label}. Note: the exact mechanism of labeling should be
+#' considered an internal implementation detail, but the label will always be retrieved via \code{obj_label}.
+#' @examples
+#' x <- formatters_with_label(c(1, 2, 3), label = "Test")
+#' obj_label(x)
+formatters_with_label <- function(x, label) {
+  obj_label(x) <- label
```
Comment (m7pr) [inline R/formatters_var_labels.R:19]: @gogonzo do you think we also need to include `obj_label` from formatters 
https://github.com/insightsengineering/formatters/blob/ffa09bb46eb28a9e269847b918c14c719c5d28b5/R/generics.R#L261C1-L283C2 ? Looks like it's a wrapper around S3 and S4 at the same time, where for S4 it uses standardGeneric and for S3 it just calls `attr(x, "label")`. I think in teal packages we focus on S3, so rewriting `obj_label(x) <- label` to `attr(x, "label") <- label` should be enough

**Comment by m7pr** [comment]
at https://github.com/insightsengineering/coredev-tasks/issues/447 we decided to move those functions to teal.data https://github.com/insightsengineering/teal.data/pull/155

**Pull request closed without merging**